base.js

```diff
@@ -81,34 +138,55 @@ function ReadResult(err, offset, value) {
     this.value = value === undefined ? null : value;
 }
 
+ReadResult.prototype.copyFrom = function copyFrom(srcResult) {
+    this.err = srcResult.err;
+    this.offset = srcResult.offset;
+    this.value = srcResult.value;
+    return this;
+};
+
+ReadResult.prototype.reset = function reset(err, offset, value) {
+    this.err = err;
+    this.offset = offset;
+    this.value = value;
+    return this;
+};
+
 ReadResult.error = function error(err, offset, value) {
-    return new ReadResult(err, offset, value);
+    throw new Error('deprecated');
```

Comment:
True, I can change this to something backwards compatible; the `throw`s were really there to assist the conversion (this is still a WIP).